packages/edit-post/src/components/meta-boxes/meta-boxes-area/index.js

```diff
@@ -6,83 +6,54 @@ import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useRef, useEffect, useState } from '@wordpress/element';
 import { Spinner } from '@wordpress/components';
-import { withSelect } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { store as editPostStore } from '../../../store';
 
-class MetaBoxesArea extends Component {
-	/**
-	 * @inheritdoc
-	 */
-	constructor() {
-		super( ...arguments );
-		this.bindContainerNode = this.bindContainerNode.bind( this );
-	}
+const MetaBoxesArea = ( { location } ) => {
+	const container = useRef( null );
+	const [ form, setForm ] = useState();
+	useEffect( () => {
+		setForm( document.querySelector( '.metabox-location-' + location ) );
 
-	/**
-	 * @inheritdoc
-	 */
-	componentDidMount() {
-		this.form = document.querySelector(
-			'.metabox-location-' + this.props.location
-		);
-		if ( this.form ) {
-			this.container.appendChild( this.form );
+		if ( form ) {
+			container.current.appendChild( form );
 		}
-	}
 
-	/**
-	 * Get the meta box location form from the original location.
-	 */
-	componentWillUnmount() {
-		if ( this.form ) {
-			document.querySelector( '#metaboxes' ).appendChild( this.form );
-		}
-	}
-
-	/**
-	 * Binds the metabox area container node.
-	 *
-	 * @param {Element} node DOM Node.
-	 */
-	bindContainerNode( node ) {
-		this.container = node;
-	}
-
-	/**
-	 * @inheritdoc
-	 */
-	render() {
-		const { location, isSaving } = this.props;
-
-		const classes = classnames(
-			'edit-post-meta-boxes-area',
-			`is-${ location }`,
-			{
-				'is-loading': isSaving,
+		return () => {
+			if ( form ) {
+				document.querySelector( '#metaboxes' ).appendChild( form );
 			}
-		);
-
-		return (
-			<div className={ classes }>
-				{ isSaving && <Spinner /> }
-				<div
-					className="edit-post-meta-boxes-area__container"
-					ref={ this.bindContainerNode }
-				/>
-				<div className="edit-post-meta-boxes-area__clear" />
-			</div>
-		);
-	}
-}
-
-export default withSelect( ( select ) => {
-	return {
-		isSaving: select( editPostStore ).isSavingMetaBoxes(),
-	};
-} )( MetaBoxesArea );
+		};
+	}, [ form ] );
```

Comment:
Maybe we should include `location` here as a dependency.

Reply:
Thanks catch!